Team Development/server/models/user.js

```diff
@@ -0,0 +1,11 @@
+var mongoose = require('mongoose');
+
+var UserSchema = new mongoose.Schema ({
+  email: {type: String, required: true},
+  password: {type: String, required: true},
+  firstName: {type: String, required: true},
+  lastName: {type: String, required: true},
+  saveList: {type: Array, required: true},
```

Comment:
saveList: Will this work for creating relationships? I think this should be: 
```javascript
saveList: [{type: mongoose.Schemea.Types.ObjectId, ref: 'Item'}]
```

Reply:
saveList shouldn't be required. This would mean that when a new user is created they would need to have a saveList at that time.

Reply:
saveList is a relationship, so absolutely should be type `mongoose.Schema.ObjectId` and reference the schema it is related to (`ref: "item"`)

Reply:
saveList should absolutely be in the form @TraceLarson described. Since it's a relationship to Item, it should be an array of objectids referencing the Item schema. Minor adjustment though. Instead of `mongoose.Schema.Types.ObjectId`, it should be `Schema.Types.ObjectId` (see [this link](https://mongoosejs.com/docs/populate.html)).

Like so:
```
saveList: [{
    type: Schema.Types.ObjectId.
    ref: "Item"
}]
```

The ref should be "Item" assuming the Item model schema is named "Item"

Reply:
I also agree that saveList shouldn't be required. A user document will start out with no item refs in the saveList.